### ISSUE TITLE
Provide ability to remove/disable TwitterUsers no longer found in Twitter

### ISSUE THREAD

**Issue by kerchner** [gwu-libraries/social-feed-manager#378]
The user_timeline job becomes inefficient when there are many TwitterUsers whose accounts are no longer active in Twitter.

I would recommend adding a --prune flag to the update_usernames management command, to remove any TwitterUsers whose id was not returned in the lookup_users call (and, of course, generate a log of the removed/disabled TwitterUsers).

**Comment by dchud** [comment]
Perhaps a "deactivate" flag rather than "prune". Prune implies deletion,
and perhaps loss of data, loss of records.

On Oct 30, 2015, at 9:38 AM, Dan Kerchner <notifications@github.com> wrote:

The user_timeline job becomes inefficient when there are many TwitterUsers
whose accounts are no longer active in Twitter.

I would recommend adding a --prune flag to the update_usernames management
command, to remove any TwitterUsers whose id was not returned in the
lookup_users call (and, of course, generate a log of the removed/disabled
TwitterUsers).

—
Reply to this email directly or view it on GitHub
<https://github.com/gwu-libraries/social-feed-manager/issues/378>.

**Comment by kerchner** [comment]
Yes, hence the title's suggested "disable" action - agreed that we wish to retain a history.  This will entail adding perhaps an "active" or similar attribute to the model for TwitterUser.

**Comment by dchud** [comment]
"is_active" is already there. :)

On Oct 30, 2015, at 9:46 AM, Dan Kerchner <notifications@github.com> wrote:

Yes, hence the title's suggested "disable" action - agreed that we wish to
retain a history. This will entail adding perhaps an "active" or similar
attribute to the model for TwitterUser.

—
Reply to this email directly or view it on GitHub
<https://github.com/gwu-libraries/social-feed-manager/issues/378#issuecomment-152530134>
.
